e2e-tests/playwright/utils/ui-helper.ts

```diff
@@ -218,6 +218,61 @@ export class UIhelper {
     await expect(elementLocator).toBeVisible();
   }
 
+  async verifyTextInSelector(selector: string, expectedText: string) {
+    const elementLocator = this.page
+      .locator(selector)
+      .getByText(expectedText, { exact: true });
+
+    try {
+      await elementLocator.waitFor({ state: "visible", timeout: 10000 });
+      const actualText = (await elementLocator.textContent()) || "No content";
+
+      if (actualText.trim() !== expectedText.trim()) {
+        console.error(
+          `Verification failed for text: Expected "${expectedText}", but got "${actualText}"`,
+        );
+        throw new Error(
+          `Expected text "${expectedText}" not found. Actual content: "${actualText}".`,
+        );
+      }
+      console.log(
+        `Text "${expectedText}" verified successfully in selector: ${selector}`,
+      );
+    } catch (error) {
+      const allTextContent = await this.page
+        .locator(selector)
+        .allTextContents();
+      console.error(
+        `Verification failed for text: Expected "${expectedText}". Selector content: ${allTextContent.join(", ")}`,
+      );
+      throw error;
+    }
+  }
+
+  async verifyPartialTextInSelector(selector: string, partialText: string) {
```

Comment:
@gustavolira this method appears to be similar to verifyTextInSelector. could we consider parameterizing verifyTextInSelector to accept exact = false? this approach might allow it to handle partial verification as well.